### DISCUSSION TITLE
Potential license violations about https://quack.im

### DISCUSSION THREAD

**Discussion by foobarz076** [SillyTavern/SillyTavern#3019] [Q&A]
> English is not my native language; please excuse typing errors. If you need to get in touch with them, I will gladly provide partial help with translation.

## In a nutshell:
I discovered a website which ，they are running some modified version of Sillytavern and have not properly provided the source code for the modified version as required by the license.

## Which website?
https://quack.im/ , it's a Chinese website which provide chatting services through a modified SillyTavern frontend.

## Where is the proof that they used modified source code?
[In their docs](https://docs.quack.im/) ,they claim to offer their services in two ways.

* "云酒馆" is just a standard hosted SillyTavern instance, while they somehow integrate the user systems on both sides.
<img width="1440" alt="Screenshot 2024-10-27 at 19 37 59" src="https://github.com/user-attachments/assets/483a5cfe-7799-479d-bce9-d96ca2594ee6">

<img width="1440" alt="Screenshot 2024-10-27 at 19 38 42" src="https://github.com/user-attachments/assets/bf12d56d-ac48-456f-8d0a-29a4b3b4022f">

> They translated ST's different character cards while they ignored creator's notes :-)

* "轻享版" (Quark Lite) is a modified interface.

<img width="1440" alt="Screenshot 2024-10-27 at 19 55 31" src="https://github.com/user-attachments/assets/fa3213ad-03b5-4a8a-8b8c-18ed8c7ca4e9">
Using the browser's developer tools, you can see that they are all called in almost the same way as SillyTavern. （For example, when click send button, it will send a `api/backends/chat-completions/generate` request to backend.

## Following up?
I requested the source code for their modified version in their chat group, but got no response.
They even removed the link I posted to the AGPL3 license used by SillyTavern and banned one of my accounts from their Discord servers.

<img width="1140" alt="Screenshot 2024-10-27 at 19 25 53" src="https://github.com/user-attachments/assets/007a8c86-55e4-4eda-948b-5f695b10101b">

<img width="1375" alt="Screenshot 2024-10-27 at 19 29 49" src="https://github.com/user-attachments/assets/45f1eb43-44e6-4bbc-9ba8-e5cb09399c35">

You may found this server's invite link through theri website, or `https://discord.gg/RBDVwJdFh3` .

**Comment by CheshireAI** [comment]
Are you sure they've actually changed anything? I run a similar commercial service with SillyTavern, and user login system is built in native to the program (see https://tavern3.openerotica.org). You can change only one line in config.yml and it works. The rate limiting is probably done with some kind of proxy like litellm.

**Reply by Cohee1207** [reply]
Tavern 3?? What happened to the other two?

**Reply by CheshireAI** [reply]
So I'm not actually sure I have this set up the most sane way. I have multiple different tavern domains, and I changed the default settings for each one so when you create a new user, the user can immediately start playing with no setup, everything is already dialed in. Most of the settings differences between instances are just stable diffusion presets with specific settings for each model and base url. Tavern1 is set up for anime styles, tavern2 is focused on furry type characters, and tavern3 is set for photorealism. I could just do one instance and let my users mess around with settings and endpoint list, but most people know nothing about AI, LLMs, or stable diffusion and get immediately intimidated when they see all the settings and terms. Managing three instances is mildly annoying, but it seems almsost perfectly optimized so my users can start gooning as fast possible with as few clicks as possible and almost no onboarding.
